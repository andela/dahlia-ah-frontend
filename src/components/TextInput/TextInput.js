import React from 'react';
import PropTypes from 'prop-types';
import './TextInput.scss';


const TextInput = (props) => {
  const {
    formField, onChange, index,
  } = props;
  return (
    <>
      <input className="text-input" type="text" placeholder={`${formField.name}`} onChange={(e) => onChange(e, index)} value={formField.value} />
      {formField.errorMessage ? <p className="validation-error">{formField.errorMessage}</p> : null }
      {/*  */}
    </>
  );
};

TextInput.propTypes = {
  formField: PropTypes.shape({
    name: PropTypes.string.isRequired,
    value: PropTypes.string.isRequired,
    errorMessage: PropTypes.string,
  }).isRequired,
  onChange: PropTypes.func.isRequired,
  index: PropTypes.number.isRequired,
};

export default TextInput;
