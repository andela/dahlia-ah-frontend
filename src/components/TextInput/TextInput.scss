.modal{
    input[type=text] {
        font-size: 16px;
        text-align: left;
    }
    
    input[type=password] {
        font-size: 16px;
        text-align: left;
    }
    
    input[type=text]:focus,
    input[type=password]:focus 
    {
        border-bottom: 1px solid rgba(0, 0, 0, 0.9) !important;
        box-shadow: 0 1px 0 0 rgba(0, 0, 0, 0.9) !important;
    }
    
    .inputBox{
        position: relative;
        margin-bottom:20px;
        span {
            display: block;
            color: red;
            font-size: 10px;
            text-align: left !important;
        }
    }
    
    label {
        position: absolute;
        top: 0;
        left: 0;
        font-size: 18px;
        pointer-events: none;
        padding: 10px 0;
        transition: .3s;
    }
    
    input[type=text]:focus ~ label,
    input[type=text]:valid ~ label,
    input[type=password]:focus ~ label,
    input[type=password]:valid ~ label,
    {
       top:-18px;
       color: #C4C4C4 !important;
       font-size: 12px;
    }
    
}

@media screen and (max-width: 420px) and (min-width: 300px) {
    .modal{
        .inputBox{
            margin-bottom:10px;
        }
    
        input[type=text] {
            font-size: 14px
        }
        
        input[type=password] {
            font-size: 14px;
        }
    
        label{
            font-size: 14px;
        }
    } 
  }