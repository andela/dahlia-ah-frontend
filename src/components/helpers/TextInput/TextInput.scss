.inputBox{
    position: relative;
    margin-bottom:10px;

    input[type=text] {
        font-size: 16px;
        text-align: left;
    }
    
    input[type=password] {
        font-size: 16px;
        text-align: left;
    }
    
    input[type=text]:focus,
    input[type=password]:focus 
    {
        border-bottom: 1px solid rgba(0, 0, 0, 0.9) !important;
        box-shadow: 0 1px 0 0 rgba(0, 0, 0, 0.9) !important;
    }
    label {
        position: absolute;
        top: 0;
        left: 0;
        font-size: 16px;
        pointer-events: none;
        padding: 10px 0;
        transition: .3s;
    }
    
    input[type=text]:focus ~ label,
    input[type=text]:valid ~ label,
    input[type=password]:focus ~ label,
    input[type=password]:valid ~ label
    {
        top:-18px;
        color: #C4C4C4 !important;
        font-size: 12px;
    }
    span {
        display: block;
        color: red;
        font-size: 10px;
        text-align: left !important;
    }
}

@media screen and (max-width: 420px) and (min-width: 300px) {
    .inputBox{
        margin-bottom:10px;
        input[type=text] {
            font-size: 14px
        }
        input[type=password] {
            font-size: 14px;
        }
        label{
            font-size: 14px;
        }
    }
  }