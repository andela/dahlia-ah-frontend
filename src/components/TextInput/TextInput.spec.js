import React from 'react';
import sinon from 'sinon';
import { configure, mount } from 'enzyme';

import Adapter from 'enzyme-adapter-react-16';
import TextInput from './TextInput';

configure({ adapter: new Adapter() });

describe('TextInput component <TextInput />', () => {
  const formField = {
    name: 'Email',
    id: 1,
    value: '',
    errorMessage: null,
  };

  it('renders correctly', () => {
    const wrapper = mount(
      <TextInput formField={formField} index={0} />,
    );

    expect(wrapper).toMatchSnapshot();
    wrapper.unmount();
  });

  it('has an input element with class text-input with correct attributes', () => {
    const wrapper = mount(
      <TextInput formField={formField} index={0} />,
    );
    expect(wrapper.find('.text-input').get(0).props.type).toEqual('text');
    expect(wrapper.find('.text-input').get(0).props.placeholder).toEqual(formField.name);
    expect(wrapper.find('.text-input').get(0).props.value).toEqual(formField.value);
    wrapper.unmount();
  });

  it('should fire onChange function when a change happens on input', () => {
    const onChange = sinon.spy();
    const wrapper = mount(
      <TextInput formField={formField} index={0} onChange={onChange} />,
    );
    wrapper.find('input').simulate('change');
    expect(onChange).toHaveProperty('callCount', 1);
    wrapper.unmount();
  });

  it('renders a p element with an error message if thers is an error message', () => {
    const errorMessage = 'invalid input';
    const wrapper = mount(
      <TextInput formField={{ ...formField, errorMessage }} index={0} />,
    );
    expect(wrapper.find('.validation-error').get(0).props.children).toEqual(errorMessage);
    wrapper.unmount();
  });
});
