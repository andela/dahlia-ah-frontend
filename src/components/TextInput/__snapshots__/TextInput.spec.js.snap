// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TextInput component <TextInput /> renders correctly 1`] = `ReactWrapper {}`;
